sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSource.scala

```diff
@@ -54,7 +58,22 @@ class StateDataSource extends TableProvider with DataSourceRegister {
       properties: util.Map[String, String]): Table = {
     val sourceOptions = StateSourceOptions.apply(session, hadoopConf, properties)
     val stateConf = buildStateStoreConf(sourceOptions.resolvedCpLocation, sourceOptions.batchId)
-    new StateTable(session, schema, sourceOptions, stateConf)
+    // Read the operator metadata once to see if we can find the information for prefix scan
+    // encoder used in session window aggregation queries. We don't need to read this if we know
+    // we are reading state for a stream-stream JOIN operator.
+    val stateStoreMetadata = if (sourceOptions.joinSide == JoinSideValues.none) {
```

Comment:
nit: Probably OK to just pass over state store metadata for all cases. Not a huge deal to optimize like so, and it's confusing when some param is effectively no op in which condition.

Reply:
Done